merlin/schema/schema.py

```diff
@@ -286,7 +286,32 @@ def apply_inverse(self, selector) -> "Schema":
         return self.excluding(selector)
 
     def select_by_tag(self, tags: Union[Union[str, Tags], List[Union[str, Tags]]]) -> "Schema":
-        """Select matching columns from this Schema object using a list of tags
+        """Select all matching columns from this Schema object using a list of tags
+
+        Parameters
+        ----------
+        tags : List[Union[str, Tags]] :
+            List of tags that describes which columns match
+
+        Returns
+        -------
+        Schema
+            New object containing only the ColumnSchemas of selected columns
+
+        """
+        if not isinstance(tags, (list, tuple)):
+            tags = [tags]
+
+        selected_schemas = {}
+
+        for _, column_schema in self.column_schemas.items():
+            if all(x in column_schema.tags for x in tags):
+                selected_schemas[column_schema.name] = column_schema
+
+        return Schema(selected_schemas)
+
+    def select_by_any_tag(self, tags: Union[Union[str, Tags], List[Union[str, Tags]]]) -> "Schema":
```

Comment:
What would you think about making the methods `select_by_all_tags` and `select_by_any_tags`, and then adding a method alias that makes `select_by_tag` refer to `select_by_any_tags` for backwards compatibility? That might be smoother than changing the behavior of the existing method.